src/hooks/useSelector.js

```diff
@@ -111,5 +92,32 @@ export function useSelector(selector, deps) {
     return () => subscription.tryUnsubscribe()
   }, [store, subscription])
 
-  return selectedState
+  try {
+    latestSelectedState.current = useMemo(() => selector(store.getState()), [
```

Comment:
I think instead of `useMemo` and the whole `id` stuff you could just capture the store state into a `latestState` ref and compare `store.getState()` to that value during render and if so, return `latestSelectedState.current`

Reply:
Oh, and assigning a ref during render is actually a side-effect, and the render function should be pure.

Reply:
Sure...

I've moved the assignment to the `latestSelectedState.current` into the `finally` clause of the hook.

Please notice that's exactly the same as executing synchronously after the return of the function, which is what `useLayoutEffect` does (in our case `useIsomorphicLayoutEffect`).

I.e: If that makes this hook impure, it also does assigning to references synchronously once the render function has finished... It's the same.

Reply:
> Oh, and assigning a ref during render is actually a side-effect, and the render function should be pure.

Ok, you are right. That was not nice. I've changed that, I've also got rid of the superfluous `useMemo`. Thanks!

Reply:
> I think instead of useMemo and the whole id stuff you could just capture the store state into a latestState ref and compare store.getState() to that value during render (and of course compare the selector) and if so, return latestSelectedState.current

Actually, I came up with a cleaner and more performant way to do that, you should check it out! I really think that my latest code makes the implementation even more understandable than it was before. :blush: